### PR TITLE
upgrade some dependencies

### PR DESCRIPTION
**Pull request by adbenitez**
we can merge this after 1.42

### CODE REVIEW

**Comment by adbenitez** [comment]
(CI failed because new gradle plugin requires java 17, I have to adapt the CI setup that is using old java)

File: build.gradle [hunk 1]
```diff
@@ -86,7 +86,7 @@ dependencies {
 android {
     namespace "org.thoughtcrime.securesms"
     flavorDimensions "none"
-    compileSdkVersion 33
+    compileSdk 34
```
Comment (adbenitez) [inline build.gradle:89]: required by webkit 1.8.0

**Comment by adbenitez** [comment]
I already had this merged in my fork since November, so far so good

File: build.gradle [hunk 2]
```diff
@@ -216,6 +214,10 @@ android {
     lint {
         abortOnError false
     }
+    buildFeatures {
+        renderScript true
+        aidl true
+    }
```
Comment (r10s) [inline build.gradle:220]: this addition added by some automation? otherwise (or in general :) - what does it mean?

File: gradle.properties [hunk 3]
```diff
@@ -1,3 +1,6 @@
+android.defaults.buildfeatures.buildconfig=true
 android.enableJetifier=true
+android.nonFinalResIds=false
+android.nonTransitiveRClass=false
```
Comment (r10s) [inline gradle.properties:4]: same here: did that come by some automation?

File: build.gradle (hunk 2 above)
Comment (adbenitez) [reply build.gradle:220]: yes, this was done automatically by AndroidStudio, it shows a dialog saying that your gradle plugin version is behind and offers to upgrade it, then it does that changes automatically, porting to the new config idioms of the new version it seems and some settings get deprecated etc

File: gradle.properties (hunk 3 above)
Comment (adbenitez) [reply gradle.properties:4]: Res IDs are non-final now in recent gradles, you probably also saw some warning in android studio where we have switch statements with branches using menu IDs etc. saying that such Res IDs are no final etc. it seems this setting here disables the dynamic Res IDs avoiding issues for our usage,

this change was done automatically by AndroidStudio

**Review by r10s — APPROVED** [review]

**Pull request merged**